controllers/release/adapter.go

```diff
@@ -119,99 +115,51 @@ func (a *Adapter) EnsureFinalizerIsAdded() (reconciler.OperationResult, error) {
 	return reconciler.ContinueProcessing()
 }
 
-// EnsureReleasePlanAdmissionEnabled is an operation that will ensure that the ReleasePlanAdmission is enabled.
-// If it is not, no further operations will occur for this Release.
-func (a *Adapter) EnsureReleasePlanAdmissionEnabled() (reconciler.OperationResult, error) {
-	_, err := a.loader.GetActiveReleasePlanAdmissionFromRelease(a.ctx, a.client, a.release)
-
-	if err != nil && strings.Contains(err.Error(), "multiple ReleasePlanAdmissions found") {
-		patch := client.MergeFrom(a.release.DeepCopy())
-		a.release.MarkInvalid(v1alpha1.ReleaseReasonValidationError, err.Error())
-		return reconciler.RequeueOnErrorOrStop(a.client.Status().Patch(a.ctx, a.release, patch))
-	}
-	if err != nil && strings.Contains(err.Error(), "auto-release label set to false") {
-		patch := client.MergeFrom(a.release.DeepCopy())
-		a.release.MarkInvalid(v1alpha1.ReleaseReasonTargetDisabledError, err.Error())
-		return reconciler.RequeueOnErrorOrStop(a.client.Status().Patch(a.ctx, a.release, patch))
-	}
-	return reconciler.ContinueProcessing()
-}
-
-// EnsureReleasePipelineRunExists is an operation that will ensure that a release PipelineRun associated to the Release
-// being processed exists. Otherwise, it will create a new release PipelineRun.
-func (a *Adapter) EnsureReleasePipelineRunExists() (reconciler.OperationResult, error) {
-	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release)
-	if err != nil && !errors.IsNotFound(err) {
-		return reconciler.RequeueWithError(err)
+// EnsureReleaseIsCompleted is an operation that will ensure that a Release is completed (marked as released) when
+// all the phases (e.g. deployment or processing) have been completed if required.
```

Comment:
reads better to say `required phases` instead of `phases .. if required`

Reply:
Changed.

Reply:
I still like the word all in there :slightly_smiling_face: `when all required phases`